### PR TITLE
feat: BSComputeShader and some improvements for BSImagespaceSahder.

### PR DESCRIPTION
**Pull request by Jonahex**

### CODE REVIEW

File: include/RE/B/BSComputeShader.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		void ReloadShaders(void* stream) override;  // 00
+
+		// add
+		virtual void Dispatch(uint32_t techniqueId, uint32_t threadGroupCountX, uint32_t threadGroupCountY, uint32_t threadGroupCountZ);  // 02
+
+		const char*                                      name;        // 18
+		BSTHashMap<uint32_t, BSGraphics::ComputeShader*> shadersMap;  // 20
+
+	private:
+		KEEP_FOR_RE()
```
Comment (FlayaN) [inline include/RE/B/BSComputeShader.h]: `KEEP_FOR_RE` is not defined for sse, only for ng, so this doesn't compile atm

File: include/RE/B/BSComputeShader.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		inline static constexpr auto VTABLE = VTABLE_BSComputeShader;
+
+		~BSComputeShader() override;  // 00
+
+		// override (BSReloadShaderI)
+		void ReloadShaders(void* stream) override;  // 00
+
+		// add
+		virtual void Dispatch(uint32_t techniqueId, uint32_t threadGroupCountX, uint32_t threadGroupCountY, uint32_t threadGroupCountZ);  // 02
+
```
Comment (FlayaN) [inline include/RE/B/BSComputeShader.h:22]: add `// members` comment above these

File: include/RE/B/BSShader.h [hunk 3]
```diff
@@ -29,27 +29,51 @@ namespace RE
 		{
 		public:
 			// members
-			std::uint32_t      id;                  // 00
-			ID3D11PixelShader* shader;              // 08
-			ConstantGroup      constantBuffers[3];  // 10
-			std::int8_t        constantTable[64];   // 58
+			std::uint32_t                id;                  // 00
+			REX::W32::ID3D11PixelShader* shader = nullptr;    // 08
```
Comment (FlayaN) [inline include/RE/B/BSShader.h]: `ID3D11PixelShader` and `ID3D11VertexShader` are forward declared in this file, this doesn't compile

File: include/RE/B/BSShader.h [hunk 4]
```diff
@@ -29,27 +29,51 @@ namespace RE
...
+			uint32_t                       unk18;           // 18
+			uint64_t                       unk20;           // 20
+			uint64_t                       unk28;           // 28
+			uint64_t                       unk30;           // 30
+			uint32_t                       unk38;           // 38
+			uint64_t                       unk40;           // 40
+			uint64_t                       unk48;           // 38
+			uint64_t                       unk50;           // 50
+			uint32_t                       unk58;           // 58
+			REX::W32::ID3D11ComputeShader* shader;          // 60
```
Comment (FlayaN) [inline include/RE/B/BSShader.h]: Forward declare `ID3D11ComputeShader` and remove namespace for this to compile

**Comment by alandtse** [comment]
Until Flayan's PR is merged, this won't compile. Don't merge yet.

**Pull request merged**